### ISSUE TITLE
Events upcoming tag not working as expected for recurring dates

### ISSUE THREAD

**Issue by mikemartin** [transformstudios/statamic-events#22]
I must have configured something incorrectly because the events tag does not return any events at the moment. See my use case here:
https://github.com/mikemartin/events-test/blob/main/resources/views/events/index.antlers.html

**Comment by grischaerbe** [comment]
I'm currently running into this as well. I'm not using the frontend part of this package but the issue might be related.
I'm using this package to write GraphQL Queries and fields for it. I can make a PR for that if you want me to.

Let's say there's an event set up like this (I'm using your provided field set):

```txt
all-day: false
multi-day: false
date-start: 2022-03-01
time-start: 14:00
time-end: 18:00
recurrence: daily
date-end: 2022-03-31
```

I'm adding a field in GraphQL to return all dates in a certain range:

```php
GraphQL::addField('Entry_Events_Event', 'datesInRange', function () {
    return [
        "type" => GraphQL::type('[String!]!'),
        'args' => [
            "start_date" => [
                "type" => GraphQL::type("String!"),
            ],
            "end_date" => [
                "type" => GraphQL::type("String!"),
            ],
        ],
        "resolve" => function (Entry $entry, $args): array {
            $event = EventFactory::createFromArray($entry->toAugmentedArray());
            $from = $args["start_date"];
            $to = $args["end_date"];
            return $event->datesBetween($from, $to)->map(function (Day $day) {
                /**
                 * TODO: This is only a debug solution
                 * add a custom return type for \TransformStudios\Events\Day
                 */
                return $day->startTime();
            })->all();
        }
    ];
});
```

A query with this field yields no dates though: 

```graphql
query DatesQuery {
  eventsEntry(
    slug: "familienfuehrung"
  ) {
    title
    datesInRange (
      start_date: "2 Mar 2022 00:53:08 GMT", 
    	end_date: "3 Mar 2022 20:53:08 GMT"
    )
  }
}
```

=> 

```json
{
  "data": {
    "eventsEntry": {
      "title": "Familienführung",
      "datesInRange": []
    }
  }
}
```

Although there should be two dates in the provided range. Stepping through the code I discovered that commenting [these lines](https://github.com/transformstudios/statamic-events-v3/blob/af3664df649a03d45c36953895df96dcad9491fe/src/Types/RecurringEvent.php#L192-L194) does return the correct dates:

```json
{
  "data": {
    "eventsEntry": {
      "title": "Ausgebucht! - Familienführung",
      "datesInRange": [
        "14:00",
        "14:00"
      ]
    }
  }
}
```

This bug only occurs when a recurring event start date is *before* `$from` in `RecurringEvent->datesBetween($from, $to)` as `$start < $after->startOfDay()` then always evaluates to `true` thus returning `null`.

**Comment by edalzell** [comment]
@adnankussair these are the kinds of issues that won't happen with the refactor.

**Comment by edalzell** [comment]
@grischaerbe I can't reproduce this, see this code:

<img width="1457" alt="CleanShot 2022-03-02 at 14 13 11@2x" src="https://user-images.githubusercontent.com/6069653/156458535-5193ca26-873e-488e-af58-43e31ca99e46.png">

**Comment by edalzell** [comment]
> 

@mikemartin that repo doesn't include the events addon, is this still occurring for you?

**Comment by edalzell** [comment]
Hey folks, I've just released v4.0, please check it out to see if it addresses your issues. Please re-open if it does.
